astropy/table/table.py

```diff
@@ -2721,7 +2721,7 @@ def to_pandas(self):
         out = OrderedDict()
 
         for name, column in self.columns.items():
-            if isinstance(column, MaskedColumn):
+            if isinstance(column, MaskedColumn) and np.any(column.mask):
                 if column.dtype.kind in ['i', 'u']:
                     out[name] = column.astype(float).filled(np.nan)
```

Comment:
Sorry for the feature creep, but maybe it's worth a warning here saying that the column is being converted to float?  This behavior is necessary but potentially confusing.

Reply:
Do you want `TableReplaceWarning` with `stacklevel=3` or just regular `AstropyUserWarning`?